vectors/no_data_vec.rs

```diff
@@ -362,14 +205,34 @@ impl<T> BorshSerialize for NoDataVec<T> {
 
 impl<T:Nondet> BorshDeserialize for NoDataVec<T> {
     fn deserialize(_buf: &mut &[u8]) -> borsh::maybestd::io::Result<Self> {
-        let res = NoDataVec::with_len(CVT_nondet_usize()) ;
+        let res = NoDataVec::new_with_len(CVT_nondet_usize()) ;
```

Comment:
this can be improved using `deserialize` code from `std::Vec`. 

We can check that input `_buf` has proper header and extract the number of elements from it.

Reply:
hmm not sure what you mean here for `no_data_vec`. For ZSTs, borsh serialize / deserialize throws an error:

```
pub const ERROR_ZST_FORBIDDEN: &str = "Collections of zero-sized types are not allowed due to deny-of-service concerns on deserialization.";
```

Reply:
`no_data_vec` is not `ZST`. It is a model of a vector for which we forget about the values of the elements, but the elements are still there.

If, for example, we deserialize a vector of 3 integers, our `NoDataVec` would be of length `3`. The size can be computed as part of deserialization. 

You can think of `NoDataVec` deserialization as deserialization of `std::Vec`, followed by creating a copy of `NoDataVec` of the same size, and then dropping `std::Vec`.